### PR TITLE
Sync APIs from karmada repo based on v1.1.0

### PR DESCRIPTION
**Pull request by riverzhang**
Signed-off-by: RongZhang [riverzhang90@gmail.com](mailto:riverzhang90@gmail.com)
What type of PR is this?
/kind api-change

What this PR does / why we need it:
Sync APIs from karmada repo based on v1.1.0.

Does this PR introduce a user-facing change?:
`Synced API based on Karmada v1.1.0`
Add `./hack/generate_client.sh` to generate karmada API Client

### CODE REVIEW

**Comment by RainbowMango** [comment]
cc @mrlihanbo @pigletfly for comments.

**Comment by riverzhang** [comment]
I need to import karmada API and Client  in my code. If not, I can't use it.

**Comment by pigletfly** [comment]
agreed to add generated client in api repo,otherwise users have to generate it themself.

**Comment by pigletfly** [comment]
should we find a way to sync apis from main karmada to here automatically when a new version is released?

**Comment by RainbowMango** [comment]
Yes, that would be great!

File: cluster/install/install.go [hunk 1]
```diff
@@ -4,8 +4,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 
-	"github.com/karmada-io/api/cluster"
-	clusterv1alpha1 "github.com/karmada-io/api/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/apis/cluster"
```
Comment (RainbowMango) [inline cluster/install/install.go:7]: Ah...
The main reason why we using the `api` repo is users don't need to inherit the whole dependencies at `karmada` repo.

Comment (riverzhang) [reply cluster/install/install.go:7]: Thanks for your comment.

**Comment by RainbowMango** [comment]
I'm hesitant to maintain the API and generated clients in two independent repos. We did that before because It only needs a few efforts to sync APIs from karmada repo and the benefit is people can import the APIs without unnecessary dependencies involved. 

In addition, from Go1.17, by leverage the [Pruned module graphs](https://go.dev/doc/go1.17) feature, the non-immediate dependencies won't block the build anymore, so the painful of import API from karmada-io/karmada has been significantly relieved.

**Comment by RainbowMango** [comment]
So, can we just sync the basic APIs for v1.1.0 this time and work out a better solution to deal with the clients after this PR?

**Comment by riverzhang** [comment]
I'm ok. I'm importing the Apis and client of the Karmada project. In theory, I should import the apis and clients of the API project. But it didn't meet users' expectations.

**Comment by RainbowMango** [comment]
>  In theory, I should import the apis and clients of the API project. But it didn't meet users' expectations.

+1
Agreed, but we have to work out the process to sync API automatically just like the [Kubenetes API](https://github.com/kubernetes/api).

**Comment by riverzhang** [comment]
Yes, it should require Karmada CI to handle this job.  Automatically handles [client-go](https://github.com/kubernetes/client-go) and [API](https://github.com/kubernetes/api) synchronization as k8S CI does.

**Comment by RainbowMango** [comment]
/hold
I'll create an issue to track the sync task. 

And this PR will be closed in favor of #10 .

**Pull request closed without merging**